lib/net.js

```diff
@@ -529,8 +529,9 @@ Socket.prototype._unrefTimer = function _unrefTimer() {
 // sent out to the other side.
 Socket.prototype._final = function(cb) {
   // If still connecting - defer handling `_final` until 'connect' will happen
-  if (this.connecting) {
+  if (this.connecting && !this._finalizingOnConnect) {
```

Comment:
I don't think this guard is need, `wriable._final()` is only called once.

Reply:
Sorry, maybe I'm missing something here :thinking: but from what I have been debugged it seems that `Socket._final()` is called for every `socket.destroySoon()` call in described scenario in the linked issue https://github.com/nodejs/node/issues/60456 

P.S. I've updated implementation to `return` from this function even if `_finalizingOnConnect` is already `true` (as it was previously).